connectors/config.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+import copy
+
+from envyaml import EnvYAML
+
+
+class ConfigNotLoadedError(Exception):
+    pass
+
+
+class Config:
+    _yaml = None
+
+    @classmethod
+    def load(cls, config_file):
+        cls._yaml = EnvYAML(config_file)
+
+    @classmethod
+    def get(cls, key=None, default=None):
+        if cls._yaml is None:
+            raise ConfigNotLoadedError(
+                "Config is not loaded yet, make sure to call Config.load(config_file) first."
+            )
+        yaml = copy.deepcopy(cls._yaml)
```

Comment:
why do you deepcopy? you can return `_yaml.get(key, default)` directly

Reply:
The `get` function can return the whole config, which is an instance of `EnvYAML`, or anything if a key is provided.

I use deepcopy to prevent tempering of the real condig data held by `Config` class. But I can remove it as anyone can directly access `Config._yaml` even I prefix `yaml` with `_`.

Reply:
ah I see the problem. It's not ideal in Python to have a function that sends back two different types of objects; this will lead to issues. Sending back copies of the yaml also breaks your goal of having a singleton I think, because you would end up with different config at the same time I suspect.

What is the use case to return the full `_yaml` ? if we have a singleton, why `BaseService` needs to call `Config.get()`? why not `Config` directly?

or even nothing since anyone needing a config can import `connectors.config` and use the singleton directly, right?

I am also now wondering, why do you need a `Config` class ?

is this not enough? 

```
_CONFIG = None 

def set_config(config_file): 
    global _CONFIG 
    _CONFIG = EnvYAML(config_file) 

def get_config():
    if _CONFIG is None:
        raise ConfigNotLoadedError() 
    return _CONFIG
```

I think `EnvYAML` has all the methods we need

Reply:
Yes, this is what I'm doing now. I don't even have the `get_config()` function, you can access the global variable `_CONFIG` directly.